cli/command/image/build_buildkit.go

```diff
@@ -228,6 +262,26 @@ func doBuild(ctx context.Context, eg *errgroup.Group, dockerCli command.Cli, opt
 			return cli.StatusError{Status: jerr.Message, StatusCode: jerr.Code}
 		}
 	}
+
+	// Everything worked so if -q was provided the output from the daemon
+	// should be just the image ID and we'll print that to stdout.
+	//
+	// TODO: we may want to use Aux messages with ID "moby.image.id" regardless of options.quiet (i.e. don't send HTTP param q=1)
+	// instead of assuming that output is image ID if options.quiet.
+	if options.quiet {
+		imageID = buf.String()
+		fmt.Fprint(dockerCli.Out(), imageID)
+	}
+
+	if options.imageIDFile != "" {
+		if imageID == "" {
+			return errors.Errorf("Server did not provide an image ID. Cannot write %s", options.imageIDFile)
```

Comment:
"cannot write image ID file, server did not provide an image ID"

Reply:
@cpuguy83 thanks! Should be fixed now!

Reply:
@cpuguy83 as in you need to use it with https://github.com/moby/moby/pull/37368